File: backend/src/main/java/com/team01/project/domain/notification/entity/NotificationList.java
package com.team01.project.domain.notification.entity;

import java.time.LocalDateTime;

import com.team01.project.domain.user.entity.User;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class NotificationList {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id", nullable = false)
	private User user;

	@Column(nullable = false)
	private String title;

	@Column(nullable = false)
	private String message;

	@Column(nullable = false)
	private LocalDateTime notificationTime; // 저장된 시간 기록

	@Builder.Default
	private boolean isRead = false;

	public NotificationList(User user, String message) {
		this.user = user;
		this.message = message;
		this.notificationTime = LocalDateTime.now();
	}

	public void markAsRead() {
		this.isRead = true;
	}
}

File: backend/src/main/java/com/team01/project/domain/notification/entity/NotificationList.kt
package com.team01.project.domain.notification.entity

import com.team01.project.domain.user.entity.User
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
data class NotificationList(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Column(nullable = false)
    val title: String,

    @Column(nullable = false)
    val message: String,

    @Column(nullable = false)
    val notificationTime: LocalDateTime,

    var isRead: Boolean = false

) {
    fun markAsRead() {
        this.isRead = true
    }

    companion object {
        fun builder() = NotificationListBuilder()
    }

    class NotificationListBuilder {
        private var id: Long? = null
        private lateinit var user: User
        private lateinit var title: String
        private lateinit var message: String
        private lateinit var notificationTime: LocalDateTime
        private var isRead: Boolean = false

        fun id(id: Long?) = apply { this.id = id }
        fun user(user: User) = apply { this.user = user }
        fun title(title: String) = apply { this.title = title }
        fun message(message: String) = apply { this.message = message }
        fun notificationTime(notificationTime: LocalDateTime) = apply { this.notificationTime = notificationTime }
        fun isRead(isRead: Boolean) = apply { this.isRead = isRead }

        fun build(): NotificationList {
            return NotificationList(
                id = id,
                user = user,
                title = title,
                message = message,
                notificationTime = notificationTime,
                isRead = isRead
            )
        }
    }
}
